src/composables/accountSelector.ts

```diff
@@ -0,0 +1,200 @@
+import {
+  computed,
+  ref,
+  toRef,
+  watch,
+} from 'vue';
+
+import type {
+  IAccountSelectorEntry,
+  ITransaction,
+} from '@/types';
+import {
+  ACCOUNT_SELECT_TYPE_FILTER,
+  AccountSelectTypeFilter,
+  PROTOCOLS,
+  TX_DIRECTION,
+} from '@/constants';
+import { createCustomScopedComposable, getDefaultAccountLabel } from '@/utils';
+import {
+  useAccounts,
+  useAddressBook,
+  useLatestTransactionList,
+  useTransactionData,
+} from '@/composables';
+
+import { useAeMiddleware } from '@/protocols/aeternity/composables';
+import { useAeNames } from '@/protocols/aeternity/composables/aeNames';
+import { getInnerTransaction } from '@/protocols/aeternity/helpers';
+
+export const useAccountSelector = createCustomScopedComposable(() => {
+  const { getName } = useAeNames();
+  const { getMiddleware } = useAeMiddleware();
+
+  const latestTransactions = ref<IAccountSelectorEntry[]>([]);
+  const searchQuery = ref<string>('');
```

Comment:
```suggestion
  const searchQuery = ref('');
```
No need to type the ref because the type is inferred from the value.